### DISCUSSION TITLE
Some guidance on deployment to Azure Container Apps please?

### DISCUSSION THREAD

**Discussion by Hypothesys-Smithgor** [microsoft/aspire#6777] [Q&A] [answered]
Hi,

I've been working my way around Aspire for the past few weeks, and can mostly get things to work (I'm working in version 9 now).
However, I've stalled when attempting to deploy to Azure Container Apps.

I have an application that works fine locally by provisioning my database as follows:

var sqldb = builder
    .AddSqlServer("sql")
    .AddDatabase("sqldata");

When I deploy this to ACA using azd, all works fine - a sql container is created.

However, I really want to use an Azure SQL Database so locally I set this as follows:

var sqldb = builder
    .AddConnectionString("sqldata");

with the following in appsettings.json

"ConnectionStrings": {
    "sqldata" : "MY CONNECTION STRING",
  }

That all works fine locally, but not when I deploy to ACA - the connection apparently doesn't get found.
I've had a look at using a KeyVault, but the documentation in the Integration section appears to be out-of-date (or I'm just not understanding it properly!).

Can anyone please advise on the steps I need to take to make this work?

Many thanks in advance.

**Comment by davidfowl** [comment, marked as answer]
This might help:

https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/integrations-overview

Edit:

https://learn.microsoft.com/en-us/dotnet/aspire/azure/integrations-overview?tabs=dotnet-cli
